portfolio/src/main/webapp/script.js

```diff
@@ -36,10 +36,32 @@ function addRandomNJPerson() {
   const personDescripContainer = document.getElementById('NJ-person-descrip');
   personNameContainer.innerText = person;
   personDescripContainer.innerText = descrip;
-  
 }
 
-function closeRandomizer(){
-  const personContainer = document.getElementById("NJ-person-container");
-  personContainer.style.display = "none"
+/**
+ * Closes the randomizer or the popup
+ */
+function closeWindow(window){
+  if (window == 'random'){
+    document.getElementById("NJ-person-container").style.display = "none";
+  } else if (window == 'popup'){
+    document.getElementById("popup").style.display = "none";
+  }
```

Comment:
This could be a lot simpler if you passed in the actual elementID. Then it would reduce to just document.getElementById(elementID)......

Alternatively you could define the mapping of random->NJ-person-container using an enum.

Embedding these mappings in logic like this is fine when it's one method, but typically is harder to maintain in the long term. In the future you might have N methods with the logic of "random" means "NJ-person-container" embedded in them. If you at some point need to change that mapping in some way, you might only notice M (where M < N) cases, and introduce more errors. Similarly, if you continue to add more popup dialogs, you'll need to keep expanding this if-structure, whereas leveraging an enum/map/etc would just require adding the additional mapping (which would be a lot easier to understand)